File: samples/FsBulletML.Sample.Unity2D.FSharp/FsBulletML.Sample.Unity2D.FSharp/BulletFunctions.fs
namespace FsBulletML.Sample.Unity2D.FSharp

open System
open System.Collections.Generic
open System.Diagnostics 
open UnityEngine
open FsBulletML

type BulletFunctions () =
  static let player = GameObject.Find("player")
  static let rand = new System.Random()

  interface IBulletMLManager with
    member this.GetRandom() = Math.Round(rand.NextDouble() * 10000.) / 10000. |> float32
    member this.GetRank () = 0.f 
    member this.GetPlayerPosX () = player.transform.position.x
    member this.GetPlayerPosY () = player.transform.position.y

File: samples/FsBulletML.Sample.Unity2D.CSharp/Assets/Scripts/BulletFunctions.cs
using UnityEngine;
using System;
using System.Collections.Generic;
using System.Diagnostics;
using FsBulletML;

public class BulletFunctions : FsBulletML.Processable.IBulletMLManager
{
    private static GameObject player;

    public BulletFunctions()
    {
        player = GameObject.Find("player");
    }

    private static System.Random rand = new System.Random();
    public float GetRandom()
    {
        return Convert.ToSingle(Math.Round(rand.NextDouble() * 10000) / 10000);
    }

    public float GetRank()
    {
        return 0;
    }

    public float GetPlayerPosX()
    {
        return player.transform.position.x;
    }

    public float GetPlayerPosY()
    {
        return player.transform.position.y;
    }
}